### PR TITLE
 Production push Master branch CI/CD

### PR DESCRIPTION
**Pull request by istiakmorsalinmlbd**

### CODE REVIEW

**Comment by istiakmorsalinmlbd** [comment]
CI build will succeed after this https://github.com/monstar-lab-bd/flutter-template/pull/2 PR being merged into master

@utpal-barman @ashraful-chowdhury

File: .github/workflows/push_main_production.yml [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+      - uses: actions/setup-java@v1
+        with:
+          java-version: '12.x'
+      - uses: subosito/flutter-action@v1
+        with:
+          channel: 'stable'
+      - run: flutter pub get
+      - run: flutter pub run build_runner build --delete-conflicting-outputs
+      - run: flutter build apk --release -t lib/main_production.dart
+      - run: flutter analyze
```
Comment (utpal-barman) [inline .github/workflows/push_main_production.yml:28]: line 27 & 28 need to be swapped, analyze should run before the build?

Comment (utpal-barman) [reply .github/workflows/push_main_production.yml:28]: `pull_request_validation.yml` needs to be changed too

Comment (istiakmorsalinmlbd) [reply .github/workflows/push_main_production.yml:28]: @utpal-barman I agree with your suggestion. But Can you please explain why it is kept like this? 
https://github.com/monstar-lab-bd/flutter-template/blob/master/.github/workflows/pull_request_validation.yml

Comment (utpal-barman) [reply .github/workflows/push_main_production.yml:28]: known issue to me, I always had to change these lines

**Review by utpal-barman — CHANGES_REQUESTED** [review]

File: .github/workflows/push_main_production.yml [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+# separate terms of service, privacy policy, and support
+# documentation.
+
+name: Push CI Production
+
+on:
+  push:
+    branches: [ "master" ]
+  pull_request:
+    branches: [ "master" ]
```
[the remark this answers is not in the document]
Comment (istiakmorsalinmlbd) [reply .github/workflows/push_main_production.yml]: Alternatively, we can remove the Pull CI. But I think your suggestions will be more appropriate

**Review by utpal-barman — APPROVED** [review]

**Pull request merged**